From b73e0fc8d40eb271548efb1dd5d02c00e8876fad Mon Sep 17 00:00:00 2001
From: Josef Hofer <me@jhofer.de>
Date: Sun, 15 Feb 2026 01:09:18 +0100
Subject: [PATCH] Add IMG_PREFIX support for custom image naming

---
 Makefile               | 4 +++-
 internal/base/pkg.yaml | 2 +-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 82c3125..25ffe8b 100644
--- a/Makefile
+++ b/Makefile
@@ -15,6 +15,7 @@ GOARCH := $(shell uname -m | sed 's/x86_64/amd64/' | sed 's/aarch64/arm64/')
 REGISTRY ?= ghcr.io
 USERNAME ?= talos-rpi5
 REGISTRY_AND_USERNAME ?= $(REGISTRY)/$(USERNAME)
+IMG_PREFIX ?=
 KRES_IMAGE ?= ghcr.io/siderolabs/kres:latest
 CONFORMANCE_IMAGE ?= ghcr.io/siderolabs/conform:latest
 
@@ -40,6 +41,7 @@ COMMON_ARGS = --file=Pkgfile
 COMMON_ARGS += --provenance=false
 COMMON_ARGS += --progress=$(PROGRESS)
 COMMON_ARGS += --platform=$(PLATFORM)
+COMMON_ARGS += --build-arg=IMG_PREFIX="$(IMG_PREFIX)"
 COMMON_ARGS += --build-arg=SOURCE_DATE_EPOCH=$(SOURCE_DATE_EPOCH)
 COMMON_ARGS += --build-arg=PKGS_PREFIX="$(PKGS_PREFIX)"
 COMMON_ARGS += --build-arg=PKGS="$(PKGS)"
@@ -135,7 +137,7 @@ reproducibility-test-local-%:  ## Builds the specified target defined in the Pkg
 
 .PHONY: $(TARGETS)
 $(TARGETS):
-	@$(MAKE) docker-$@ TARGET_ARGS="--tag=$(REGISTRY_AND_USERNAME)/$@:$(IMAGE_TAG) --push=$(PUSH)"
+	@$(MAKE) docker-$@ TARGET_ARGS="--tag=$(REGISTRY_AND_USERNAME)/$(IMG_PREFIX)$@:$(IMAGE_TAG) --push=$(PUSH)"
 
 .PHONY: deps.png
 deps.png:  ## Generates a dependency graph of the Pkgfile.
diff --git a/internal/base/pkg.yaml b/internal/base/pkg.yaml
index d5a65c0..414d9cb 100644
--- a/internal/base/pkg.yaml
+++ b/internal/base/pkg.yaml
@@ -4,7 +4,7 @@ shell: /bin/bash
 dependencies:
   - image: "{{ .BUILD_ARG_TOOLS_PREFIX }}/tools:{{ .BUILD_ARG_TOOLS }}"
     to: /rootfs
-  - image: "{{ .BUILD_ARG_PKGS_PREFIX }}/kernel:{{ .BUILD_ARG_PKGS }}"
+  - image: "{{ .BUILD_ARG_PKGS_PREFIX }}/{{ .BUILD_ARG_IMG_PREFIX }}kernel:{{ .BUILD_ARG_PKGS }}"
     from: /dtb
     to: /rootfs/dtb
 finalize:
-- 
2.52.0

